rem/DB.swift

```diff
@@ -268,6 +304,57 @@ class DatabaseManager {
         return results
     }
     
+    func searchFilteredByAppName(appName: String, limit: Int = 9, offset: Int = 0) -> [(frameId: Int64, fullText: String, applicationName: String?, timestamp: Date, filePath: String, offsetIndex: Int64)] {
+        let query = frames
+            .join(videoChunks, on: frames[chunkId] == videoChunks[id])
+            .join(uniqueAppNames, on: uniqueAppNames[activeApplicationName] == frames[activeApplicationName])
+            .filter(uniqueAppNames[activeApplicationName].lowercaseString == appName.lowercased())
+            .select(frames[id], frames[activeApplicationName], frames[timestamp], videoChunks[filePath], frames[offsetIndex])
+            .limit(limit, offset: offset)
+        
+        var results: [(Int64, String, String?, Date, String, Int64)] = []
+        do {
+            for row in try db.prepare(query) {
+                let frameId = row[frames[id]]
+                let applicationName = row[frames[activeApplicationName]]
+                let timestamp = row[frames[timestamp]]
+                let filePath = row[videoChunks[filePath]]
+                let offsetIndex = row[frames[offsetIndex]]
+                results.append((frameId, "", applicationName, timestamp, filePath, offsetIndex))
+            }
+        } catch {
+            print("Search error: \(error)")
+        }
+        return results
+    }
+
+    func searchFilteredByAppNameAndText(appName: String, searchText: String, limit: Int = 9, offset: Int = 0) -> [(frameId: Int64, fullText: String, applicationName: String?, timestamp: Date, filePath: String, offsetIndex: Int64)] {
+        let query = allText
+            .join(frames, on: frames[id] == allText[frameId])
+            .join(videoChunks, on: frames[chunkId] == videoChunks[id])
+            .join(uniqueAppNames, on: uniqueAppNames[activeApplicationName] == frames[activeApplicationName])
```

Comment:
seems like this is the only line that is changed from the above function - can we just add this conditionally in both `search` and getRecentResults specified by an optional parameter instead?

I think it would be this single line and a change to the function signature instead - saving like 50 lines of code

Reply:
like:

```swift
var partialQuery = allText
            .join(frames, on: frames[id] == allText[frameId])
            .join(videoChunks, on: frames[chunkId] == videoChunks[id])

if appName != "" {
    partialQuery = partialQuery.join(uniqueAppNames, on: uniqueAppNames[activeApplicationName] == frames[activeApplicationName])
}

let query = partialQuery
            .filter(text.match("*\(searchText)*") && uniqueAppNames[activeApplicationName].lowercaseString == appName.lowercased())
            .select(allText[frameId], text, frames[activeApplicationName], frames[timestamp], videoChunks[filePath], frames[offsetIndex])
            .limit(limit, offset: offset)
```